cmd/csi_driver/main.go

```diff
@@ -42,13 +55,203 @@ var (
+	}
+	currPort := strings.TrimSpace(string(file))
+	expectedPortStr := strconv.Itoa(expectedPort)
+	if currPort != expectedPortStr {
+		klog.V(3).Infof("Your node already has Lustre kernel modules installed with an outdated configuration. Please upgrade your node pool to apply the correct settings.")
+
+		return fmt.Errorf("lnet port mismatched, expected %s, but found: %s", expectedPortStr, currPort)
+	}
+
+	log.Println("LNET_PORT matches configuration. Check passed.")
+
+	return nil
+}
+
+func configureRoute(nicName string, tableID int) error {
+	link, err := netlink.LinkByName(nicName)
+	if err != nil {
+		return fmt.Errorf("could not get link for %s: %w", nicName, err)
+	}
+	addrs, err := netlink.AddrList(link, netlink.FAMILY_V4)
+	if err != nil || len(addrs) == 0 {
+		return fmt.Errorf("could not get address for %s: %w", nicName, err)
+	}
+	klog.Infof("IP address are: %+v", addrs)
+	ipAddr := addrs[0].IP
+	if err := configureRoutesForNic(nicName, ipAddr, tableID); err != nil {
+		return fmt.Errorf("failed to configure routes for %s: %w", nicName, err)
+	}
+
+	return nil
+}
+
+func kmodInstaller() error {
+	if err := checkLnetPort(getLnetPort()); err != nil {
+		return err
+	}
+	nics, err := getGvnicNames()
+	if err != nil {
+		return fmt.Errorf("error getting nic names: %w", err)
+	}
+	if len(nics) == 0 {
+		return fmt.Errorf("no nics with eth prefix found")
+	}
+	isMultiNic, err := isMultiRailEnabled()
+	if err != nil {
+		return err
+	}
+	klog.V(4).Infof("Is using LustreCSI Multi-Rail feature: %v", isMultiNic)
+
+	lnetArg := fmt.Sprintf(`lnet.networks="tcp0(%s)"`, nics[0])
+	if isMultiNic {
+		lnetArg = fmt.Sprintf(`lnet.networks="tcp0(%s)"`, strings.Join(nics, ","))
+	}
+
+	cmd := exec.Command("/usr/bin/cos-dkms", "install", "lustre-client-drivers",
+		"--gcs-bucket=cos-default",
+		"--latest",
+		"-w", "0",
+		"--kernelmodulestree=/host_modules",
+		"--module-arg="+lnetArg,
+		"--module-arg=lnet.accept_port="+strconv.Itoa(getLnetPort()),
+		"--lsb-release-path=/host_etc/lsb-release",
+		"--insert-on-install",
+		"--logtostderr")
+
+	klog.Infof("Executing command: %s", cmd.String())
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("command execution failed: %w\noutput:\n%s", err, string(output))
+	}
+
+	klog.Infof("Command output:\n%s\n", string(output))
+	// Table ID may need to change in the future. We currently increment for every nic route after 100.
+	tableID := 100
+	if isMultiNic {
+		// Configure route for all NICS
+		for _, nicName := range nics {
+			err = configureRoute(nicName, tableID)
+			if err != nil {
+				return err
+			}
+			tableID++
+		}
+	} else {
+		// Configure route for one NIC.
+		err = configureRoute(nics[0], tableID)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
 func main() {
 	klog.InitFlags(nil)
 	flag.Parse()
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
+	if err := kmodInstaller(); err != nil {
```

Comment:
We should use a flag to control whether to install the Lustre kernel modules. Some OSS users may prefer to install and manage the Lustre kmod independently from the CSI driver.